clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -35,13 +35,12 @@ public InFlightRequests(int maxInFlightRequestsPerConnection) {
     /**
      * Add the given request to the queue for the connection it was directed to
      */
-    public void add(ClientRequest request, long now) {
+    public void add(ClientRequest request) {
         Deque<ClientRequest> reqs = this.requests.get(request.request().destination());
         if (reqs == null) {
-            reqs = new ArrayDeque<ClientRequest>();
+            reqs = new ArrayDeque<>();
             this.requests.put(request.request().destination(), reqs);
         }
-        request.setSendMs(now);
```

Comment:
It means that you don't need to pass the `now` parameter across 3 layers just for `add` to do what could be done in `doSend`. Also as Jason pointed out in the original change, it's a bit weird for `InFlightRequests` to mutate the request in this way.